vcpkg/concepts/registries.md

```diff
@@ -3,49 +3,404 @@ title: Registries concepts
+the port name followed by a hyphen. For example, the versions file for port
+`foo` must be stored in `versions/f-/foo.json`.
+
+The purpose of the versions file is two-fold:
+
+* List all available versions of each port
+* Point to the retrieval locations of each of these versions.
+
+The layout of the version file is an object containing a `"versions"` array, with
+each entry in that array being a version object.  A version object must contain
+the following properties: 
+
+* A version property: The property's key and value must match the ones used by
+  the port in its `vcpkg.json` file. The key must be one of `version`,
+  `version-semver`, `version-date`, or `version-string`; the value must be the
+  version as it appears in the port's manifest file (`vcpkg.json`). 
+* `port-version`: the value is the port's `port-version` as it appears in the
+  port's `vcpkg.json` file. 
+* `git-tree`: (only on Git registries) the value is the git-tree SHA
+  corresponding to the port's directory. This is a SHA calculated by hashing the
+  contents of the port's directory; this git-tree SHA can be used by Git to
+  retrieve the contents of the port matching the provided git-tree. This makes
+  it possible for vcpkg to retrieve old versions of ports from the registries
+  Git history. Read the [git registries](#git-registries) section to learn how
+  to obtain this value for any given port version.
+* `path`: (only on filesystem registries) the value is the full path to a
+  directory containing the port files for the specific version.
+
+## Example of a filesystem registry version file
+
+```json
+{
+  "versions": [
+    {
+      "path": "$/ports/foo/1.2.0",
+      "version": "1.2.0",
+      "port-version": 0
+    },
+    {
+      "path": "$/ports/foo/1.1.0",
+      "version": "1.1.0",
+      "port-version": 0
+    },
+    {
+      "path": "$/ports/foo/1.0.0",
+      "version": "1.0.0",
+      "port-version": 0
+    }
+  ]
+}
+``` 
+
+See the reference documentation for:
+
+* version file layout for [Git
+  registries](../maintainers/registries.md#git-version-file)
+* version file layout for [filesystem
+  registries](../maintainers/registries.md#filesystem-version-file)
 
 ## Built-in registry
 
-The built-in registry refers to the implicit registry typically used in classic
-mode scenarios, and edited directly in the directory `VCPKG_ROOT`.
+The built-in registry is a special kind of registry. The built-in registry
+is the default registry used in some operation modes, like [classic
+mode](classic-mode.md). When a [default
+registry](../reference/vcpkg-configuration-json.md#default-registry) is not
+specified, vcpkg implicity uses the built-in registry as the default.
 
-If vcpkg was acquired using `git clone`, this will refer to the registry in `VCPKG_ROOT` itself,
-and is expected to be a clone of <https://github.com/Microsoft/vcpkg> created before running vcpkg.
+The built-in registry most often refers to the local copy of the curated
```

Comment:
You changed this to 'most often', under what circumstances is it not this?

Reply:
```suggestion
The built-in registry refers to the local copy of the curated
```

Reply:
There are `--x-builtin-[ports|versions]-dir` redirection arguments that can change that.